ifsbench/benchmark.py

```diff
@@ -0,0 +1,194 @@
+# (C) Copyright 2020- ECMWF.
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+# In applying this licence, ECMWF does not waive the privileges and immunities
+# granted to it by virtue of its status as an intergovernmental organisation
+# nor does it submit to any jurisdiction.
+
+"""
+Generic benchmark implementation.
+"""
+
+from dataclasses import dataclass, field
+import os
+from pathlib import Path
+from typing import List, Optional
+
+from ifsbench.application import Application
+from ifsbench.arch import Arch
+from ifsbench.data import DataHandler
+from ifsbench.env import EnvHandler, DefaultEnvPipeline
+from ifsbench.job import Job
+from ifsbench.launch import Launcher
+
+
+__all__ = ['DefaultScienceSetup', 'DefaultTechSetup', 'DefaultBenchmark']
+
+@dataclass
+class DefaultScienceSetup:
```

Comment:
All the classes here are called DefaultSomething but don't implement a generic base class; is the plan to have other implementations? Should the base class already be defined then? Otherwise, can we remove the default for now? If other implementations are added it will anyway be a smaller refactor if all the references do not include the Default prefixed reference but the generic name.